assets/js/query/term.ts

```diff
@@ -0,0 +1,90 @@
+import { MatcherFactory } from './matcher';
+
+import { numberFields, dateFields, literalFields, termSpaceToImageField, defaultField } from './fields';
+import { FieldName, FieldMatcher, RangeEqualQualifier, TermType, AstMatcher } from './types';
+
+type RangeInfo = [FieldName, RangeEqualQualifier, TermType];
+
+function normalizeTerm(term: string, wildcardable: boolean) {
+  if (!wildcardable) {
+    return term.replace('\\"', '"');
+  }
+  return term.replace(/\\([^*?])/g, '$1');
+}
+
+function parseRangeField(field: string): RangeInfo | null {
+  if (numberFields.indexOf(field) !== -1) {
+    return [field, 'eq', 'number'];
+  }
+
+  if (dateFields.indexOf(field) !== -1) {
+    return [field, 'eq', 'date'];
+  }
+
+  const qual = /^(\w+)\.([lg]te?|eq)$/.exec(field);
+
+  if (qual) {
+    const fieldName: FieldName = qual[1];
+    const rangeQual = qual[2] as RangeEqualQualifier;
+
+    if (numberFields.indexOf(fieldName) !== -1) {
+      return [fieldName, rangeQual, 'number'];
+    }
+
+    if (dateFields.indexOf(fieldName) !== -1) {
+      return [fieldName, rangeQual, 'date'];
+    }
+  }
+
+  return null;
+}
+
+function makeTermMatcher(term: string, fuzz: number, factory: MatcherFactory): [FieldName, FieldMatcher] {
+  let rangeParsing, candidateTermSpace, termCandidate;
+  let localTerm = term;
+  const wildcardable = fuzz === 0 && !/^"([^"]|\\")+"$/.test(localTerm);
+
+  if (!wildcardable && !fuzz) {
+    // Remove quotes around quoted literal term
+    localTerm = localTerm.substring(1, localTerm.length - 1);
+  }
+
+  localTerm = normalizeTerm(localTerm, wildcardable);
+
+  // N.B.: For the purposes of this parser, boosting effects are ignored.
+  const matchArr = localTerm.split(':');
+
+  if (matchArr.length > 1) {
+    candidateTermSpace = matchArr[0];
+    termCandidate = matchArr.slice(1).join(':');
+    rangeParsing = parseRangeField(candidateTermSpace);
+
+    if (rangeParsing) {
+      const [fieldName, rangeType, fieldType] = rangeParsing;
+
+      if (fieldType === 'date') {
+        return [fieldName, factory.makeDateMatcher(termCandidate, rangeType)];
+      }
+
+      return [fieldName, factory.makeNumberMatcher(parseFloat(termCandidate), fuzz, rangeType)];
+    }
+    else if (literalFields.indexOf(candidateTermSpace) !== -1) {
+      return [candidateTermSpace, factory.makeLiteralMatcher(termCandidate, fuzz, wildcardable)];
+    }
+    else if (candidateTermSpace === 'my') {
+      return [candidateTermSpace, factory.makeUserMatcher(termCandidate)];
+    }
+  }
+
+  return [defaultField, factory.makeLiteralMatcher(localTerm, fuzz, wildcardable)];
+}
+
+export function parseTerm(term: string, fuzz: number, factory: MatcherFactory): AstMatcher {
```

Comment:
I would name this something like `getAstMatcherForTerm` as it makes it clearer what is the end result